ops/lib.rs

```diff
@@ -910,22 +910,33 @@ fn codegen_u8_slice(core: &TokenStream2, idx: usize) -> TokenStream2 {
     let value = args.get(#idx as i32);
     match #core::v8::Local::<#core::v8::ArrayBuffer>::try_from(value) {
       Ok(b) => {
-        let store = b.data() as *mut u8;
-        // SAFETY: rust guarantees that lifetime of slice is no longer than the call.
-        unsafe { ::std::slice::from_raw_parts_mut(store, b.byte_length()) }
+        // Handles detached buffers.
+        let byte_length = b.byte_length();
+        if byte_length == 0 {
+          &mut []
+        } else {
+          let store = b.data() as *mut u8;
+          // SAFETY: rust guarantees that lifetime of slice is no longer than the call.
+          unsafe { ::std::slice::from_raw_parts_mut(store, byte_length) }
+        }
```

Comment:
oh it does? TIL.

Reply:
> oh it does? TIL.

Oh, I don't know whether `.data()` throws if it's detached, or anything of that sort though, because that's why it was originally special-cased in JavaScript.
I don't believe that `.data()` is checked at all, but do confirm this yourself.

Reply:
seems to work

![image](https://user-images.githubusercontent.com/34997667/201275956-cf8b17e5-3ac8-4b36-a2c2-4296e0cb51ab.png)